pkg/storage/sync/storage.go

```diff
@@ -0,0 +1,148 @@
+package sync
+
+import (
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	meta "github.com/weaveworks/ignite/pkg/apis/meta/v1alpha1"
+	"github.com/weaveworks/ignite/pkg/client"
+	"github.com/weaveworks/ignite/pkg/storage"
+	"github.com/weaveworks/ignite/pkg/storage/watch"
+	"github.com/weaveworks/ignite/pkg/storage/watch/update"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// SyncStorage is a Storage implementation taking in multiple Storages and
+// keeping them in sync. Any write operation executed on the SyncStorage
+// is propagated to all of the Storages it manages (including the embedded
+// one). For any retrieval or generation operation, the embedded Storage
+// will be used (it is treated as read-write). As all other Storages only
+// receive write operations, they can be thought of as write-only.
+type SyncStorage struct {
+	storage.Storage
+	storages    []storage.Storage
```

Comment:
rename to writeOnlyStorages or similar?

Reply:
The `storages` slice contains _all_ storages, so both the write-only ones as well as the read-write one (which is embedded as well). Basically every storage that the writes get propagated to.

Reply:
```golang
storages:     append(wStorages, rwStorage),
```